torch/csrc/jit/passes/shape_analysis.cpp

```diff
@@ -529,10 +540,20 @@ class ShapePropagator {
         return;
       }
       case aten::_unwrap_optional: {
+        // if we have None as input, we need to leave the output alone
         auto input_ivalue = toIValue(node->input());
         if (input_ivalue && input_ivalue->isNone()) {
           return;
         }
+	// During analysis we don't want to pass None through here
+	// to not mess up the expectation of the passes working
+	// with the nodes consuming the output.
+        if (auto ot = node->input()->type()->cast<OptionalType>()) {
+          node->output()->setType(ot->getElementType());
+        } else if (!node->input()->type()->isSubtypeOf(NoneType::get())) {
```

Comment:
Nit: Maybe rewrite `!node->input()->type()->cast<NoneType>` so that it follows a little more closely with your previous check

Reply:
I think if we use mustBeNone, we have both covered, so this can be a plain else.